### ISSUE TITLE
Loading of `compile-metadata` seems slower than before.

### ISSUE THREAD

**Issue by ta0kira** [ta0kira/zeolite#179]
This is likely due to the file becoming larger, e.g., for `lib/util`, rather than a change in how loading is done.

**Comment by ta0kira** [comment]
It could also be the freshness checks, which examine the modification times of all files.

It's worth making various changes to `lib/util/.zeolite-cache/compile-metadata` to see where the bottleneck is.

**Comment by ta0kira** [comment]
It looks like the loading itself is fine, but something after that is slow. Perhaps related to #180, where `IO` actions are inadvertently getting executed multiple times.

**Comment by ta0kira** [comment]
I just remembered that sometimes when a single module is out-of-date (e.g., `base`) there will be numerous errors about it. This could mean that the freshness checks are happening multiple times for the same dependency during loading.

**Comment by ta0kira** [comment]
I did a few tests and found that nothing seems to be getting executed multiple times, but several modules have 100+ individual files to check for freshness, e.g., `lib/util` with 130+ files listed in its `compile-metadata`. All of the checks happen after the messages about loading the individual modules, which makes the delay more noticeable.

I also noticed that when forcing (`-f`) the modules seem to get loaded twice if everything seems to be up-to-date already. But, I don't think that's anything to worry about at the moment.
